### ISSUE TITLE
Save to audio file seemingly not working for large texts 

### ISSUE THREAD

**Issue by CounterFlow64** [mkiol/dsnote#47]
I must first say that this project is amazing, really a game changer for me since I don't need to fiddle with conda environments in terminals to get different models working.

I am right now trying to transcribe a book with about 700 pages, since there is no audio book version, and especially the Piper Joe Medium model sounded amazing.

But it just doesn't save. It does though if I cut it in smaller chunks. I tried wav and Opus, thinking compression might have broke it, but nothing seems to make it save. It outputs a initialization error. "Error: text to speech initialization engine has failed"

Also, it refuses to initialize TTS again afterwards, and the app needs a restart.

I am on a fedora linux 38 system. I'm using the latest version of Speech Note.

Here are outputs from terminal upon trying to save the Wav file:

![image](https://github.com/mkiol/dsnote/assets/79510944/4e7d258c-f382-48d9-b9fc-bdb8973e27ce)

Same colorful text all over til the very end. 

Interestingly Vorbis had the same pattern, but something different at the very end:

![image](https://github.com/mkiol/dsnote/assets/79510944/c9bdcd91-523f-442b-8fe5-4c1f77b7a90a)

**Comment by mkiol** [comment]
Thanks for the report.

Could you please run the app with `--verbose` option?

Few things:

- Opus is not supported yet, only Ogg Vorbis
- 700 pages?? That's a lot. Maybe you are short on disk space? Even if you use compression, app splits text into sentences and caches them in wav format. Cache is cleared on app start.
- Did you try beta version on [`flatpak-beta`](https://discourse.flathub.org/t/how-to-use-flathub-beta/2111)? Recently I've fixed few bugs in audio encoder/decoder.

**Comment by CounterFlow64** [comment]
Hello, this is the output with `--verbose `

```
[wav @ 0x56042de6b6c0] Packet corrupt (stream = 0, dts = NOPTS).
[D] 20:47:26.173 0x7f1ee2045d80 init_av_in_format:301 - opening file: /home/linuxworkstation/.var/app/net.mkiol.SpeechNote/cache/net.mkiol/dsnote/15696626062134395605.wav
[W] 20:47:26.173 0x7f1ee2045d80 () - compressor error: avformat_open_input error
[D] 20:47:26.230 0x7f1ee2045d80 () - cancel
[D] 20:47:26.230 0x7f1ee2045d80 () - stop tts engine
[D] 20:47:26.230 0x7f1ee2045d80 request_stop:126 - tts stop requested
[D] 20:47:26.230 0x7f1ee2045d80 () - service refresh status, new state: idle
[D] 20:47:26.230 0x7f1ee2045d80 () - service state changed: writing-speech-to-file => idle
[D] 20:47:26.230 0x7f1ee2045d80 () - service refresh status, new state: idle
[D] 20:47:26.230 0x7f1ee2045d80 () - service refresh status, new state: idle
[D] 20:47:26.230 0x7f1ee2045d80 () - task state changed: 2 => 0
[D] 20:47:26.230 0x7f1ee2045d80 () - tts engine error
[D] 20:47:26.230 0x7f1ee2045d80 () - transcribe task has finished
[W] 20:47:26.230 0x7f1ee2045d80 () - invalid task, reseting task state
[D] 20:47:26.230 0x7f1ee2045d80 () - app service state: writing-speech-to-file => idle
[W] 20:47:26.232 0x7f1ee2045d80 () - no available tts models for out mnt
[W] 20:47:26.232 0x7f1ee2045d80 () - invalid task, reseting task state
[W] 20:47:26.232 0x7f1ee2045d80 () - ignore TaskStatePropertyChanged signal
[D] 20:47:26.239 0x7f1ee2045d80 () - service error: tts-engine-error
```
I named Vorbis as Opus accidentally, I know it's not supported

I am not short on disk space, I have terabytes left 

I will try the beta now 👍

**Comment by CounterFlow64** [comment]
It appears beta didn't change anything, and returns the same error.

```
[wav @ 0x560ca9a9c780] Packet corrupt (stream = 0, dts = NOPTS).
[D] 20:58:14.101 0x7f9f59e41d80 init_av_in_format:301 - opening file: /home/linuxworkstation/.var/app/net.mkiol.SpeechNote/cache/net.mkiol/dsnote/14958908916168318381.wav
[W] 20:58:14.101 0x7f9f59e41d80 () - compressor error: avformat_open_input error
[D] 20:58:14.157 0x7f9f59e41d80 () - cancel
[D] 20:58:14.157 0x7f9f59e41d80 () - stop tts engine
[D] 20:58:14.157 0x7f9f59e41d80 request_stop:104 - tts stop requested
[D] 20:58:14.157 0x7f9f59e41d80 () - service refresh status, new state: idle
[D] 20:58:14.157 0x7f9f59e41d80 () - service state changed: writing-speech-to-file => idle
[D] 20:58:14.157 0x7f9f59e41d80 () - service refresh status, new state: idle
[D] 20:58:14.157 0x7f9f59e41d80 () - service refresh status, new state: idle
[D] 20:58:14.157 0x7f9f59e41d80 () - task state changed: 2 => 0
[D] 20:58:14.157 0x7f9f59e41d80 () - tts engine error
[D] 20:58:14.157 0x7f9f59e41d80 () - transcribe task has finished
[W] 20:58:14.157 0x7f9f59e41d80 () - invalid task, reseting task state
[D] 20:58:14.157 0x7f9f59e41d80 () - app service state: writing-speech-to-file => idle
[W] 20:58:14.161 0x7f9f59e41d80 () - no available tts models for out mnt
[W] 20:58:14.161 0x7f9f59e41d80 () - invalid task, reseting task state
[W] 20:58:14.164 0x7f9f59e41d80 () - ignore TaskStatePropertyChanged signal
[D] 20:58:14.164 0x7f9f59e41d80 () - service error: tts-engine-error

```

**Comment by mkiol** [comment]
Thank you for the logs.

Would you be able to share this wav file mentioned in the error log?

```
[D] 20:58:14.101 0x7f9f59e41d80 init_av_in_format:301 - opening file: /home/linuxworkstation/.var/app/net.mkiol.SpeechNote/cache/net.mkiol/dsnote/14958908916168318381.wav
```

**Comment by CounterFlow64** [comment]
Yes, but I restarted the app so it isn't the exact same. I used the least demanding robot voice so it came out fast.

Wav seems not to be supported on github, so I guess a cloud link will work. It's the Wav file from the same corresponding error.

https://drive.proton.me/urls/KENBGT29JR#YoaH2Pd9ADR9

**Comment by mkiol** [comment]
Thanks for the file and many thanks for catching this bug.

Fix: 960fef97045392e3708c509db511a7f5e26525e2

I will push this patch to `beta` channel today.

**Comment by CounterFlow64** [comment]
> Thanks for the file and many thanks for catching this bug.
> 
> Fix: [960fef9](https://github.com/mkiol/dsnote/commit/960fef97045392e3708c509db511a7f5e26525e2)
> 
> I will push this patch to `beta` channel today.

Thanks a lot!
